mm2src/coins/eth/eth_balance_events.rs

```diff
@@ -0,0 +1,170 @@
+use async_trait::async_trait;
+use common::{executor::{AbortSettings, SpawnAbortable, Timer},
+             log, Future01CompatExt};
+use futures::{channel::oneshot::{self, Receiver, Sender},
+              stream::FuturesUnordered,
+              StreamExt};
+use instant::Instant;
+use mm2_core::mm_ctx::MmArc;
+use mm2_err_handle::prelude::MmError;
+use mm2_event_stream::{behaviour::{EventBehaviour, EventInitStatus},
+                       Event, EventStreamConfiguration};
+use mm2_number::BigDecimal;
+use std::collections::HashMap;
+
+use super::EthCoin;
+use crate::{eth::{u256_to_big_decimal, Erc20TokenInfo},
+            BalanceError, MmCoin};
+
+/// This implementation differs from others, as they immediately return
+/// an error if any of the requests fails. This one completes all futures
+/// and returns their results individually.
+async fn get_all_balance_results_concurrently(
+    coin: &EthCoin,
+) -> Vec<Result<(String, BigDecimal), (String, MmError<BalanceError>)>> {
+    let mut tokens = coin.get_erc_tokens_infos();
+
+    // Workaround for performance purposes.
+    //
+    // Unlike tokens, the platform coin length is constant (=1). Instead of creating a generic
+    // type and mapping the platform coin and the entire token list (which can grow at any time), we map
+    // the platform coin to Erc20TokenInfo so that we can use the token list right away without
+    // additional mapping.
+    tokens.insert(coin.ticker.clone(), Erc20TokenInfo {
+        token_address: coin.my_address,
+        decimals: coin.decimals,
+    });
+
+    let jobs = tokens
+        .into_iter()
+        .map(|(token_ticker, info)| async move { fetch_balance(coin, token_ticker, &info).await })
+        .collect::<FuturesUnordered<_>>();
+
+    jobs.collect().await
+}
+
+async fn fetch_balance(
+    coin: &EthCoin,
+    token_ticker: String,
+    info: &Erc20TokenInfo,
+) -> Result<(String, BigDecimal), (String, MmError<BalanceError>)> {
+    let (balance_as_u256, decimals) = if token_ticker == coin.ticker {
+        (
+            coin.address_balance(coin.my_address)
+                .compat()
+                .await
+                .map_err(|e| (token_ticker.clone(), e))?,
+            coin.decimals,
+        )
+    } else {
+        (
+            coin.get_token_balance_by_address(info.token_address)
+                .await
+                .map_err(|e| (token_ticker.clone(), e))?,
+            info.decimals,
+        )
```

Comment:
Since this PR will be merged before https://github.com/KomodoPlatform/komodo-defi-framework/pull/1979
This just a comment for me as a reminder to implement balance streaming for all addresses in the HD wallet.